PUML.h

```diff
@@ -44,7 +44,78 @@ enum DataType
 	VERTEX = 1
 };
 
-#define checkH5Err(...) _checkH5Err(__VA_ARGS__, __FILE__, __LINE__)
+/**
+ * Distributes a number of mesh entities (i.e. elements or vertices) to  a given number of ranks
+ * For E entities and R ranks, the first E%R ranks will read E/R+1 entities.
+ * The remaining ranks will read E/R entities.
+ */
+class Distributor {
+  public: 
+  Distributor() = default;
+  Distributor(unsigned long numEntities, unsigned long numRanks) {
+    init(numEntities, numRanks);
+  }
+
+  void init(unsigned long newNumEntities, unsigned long newNumRanks) {
+    assert(newNumEntities > newNumRanks); 
+    numEntities = newNumEntities;
+    numRanks = newNumRanks;
+    entitiesPerRank = numEntities / numRanks;
+    missingEntities = numEntities % numRanks;
```

Comment:
Why is this not in the constructor?
The construcor + init() pattern is very common in our codebase but is actually not a great design..

Reply:
If you wouldn't use it, everything here could be const

Reply:
I've checked below, it can be changed!